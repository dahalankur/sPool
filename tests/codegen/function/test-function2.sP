# testing idempotence while using store
shared int y = 100

def store int getY(int a):
    return y
;

println(int_to_string(getY(5)))
y = 200
println(int_to_string(getY(5))) # aha! this will still return 100, which was cached in store


def quack hi():
    println(int_to_string(getY(6))) # only this will return 200 because param 6 is not cached
    println(int_to_string(getY(5))) # still 100!
    return
;

hi()