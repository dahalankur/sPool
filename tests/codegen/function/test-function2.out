100
100
200
100
